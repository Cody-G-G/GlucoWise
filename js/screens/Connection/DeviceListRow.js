'use strict';
import React, {Component} from 'react';
import styles from "./styles";
import {StyleSheet, View, Text, TouchableOpacity} from 'react-native';
import TextBold from "../../helpers/components/TextBold";

export default class DeviceListRow extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        const device = JSON.parse(this.props.device);
        const deviceButtonStyle = StyleSheet.flatten([styles.deviceButton, {backgroundColor: this.buttonColor(device)}]);

        return (
            <View style={styles.device}>
                <Text style={styles.deviceDescription}>
                    <TextBold>Name:</TextBold> {device.name}{"\n"}
                    <TextBold>Id:</TextBold> {device.id}
                </Text>

                <TouchableOpacity style={deviceButtonStyle} onPress={() => this.props.onPress(device)}>
                    <Text style={styles.deviceButtonText}>{this.buttonText(device)}</Text>
                </TouchableOpacity>
            </View>
        );
    }

    buttonColor(device) {
        return this.props.connectedDeviceIDs.includes(device.id) ? 'firebrick' : 'green';
    };

    buttonText(device) {
        return this.props.connectedDeviceIDs.includes(device.id) ? "Disconnect" : "Connect";
    }
}