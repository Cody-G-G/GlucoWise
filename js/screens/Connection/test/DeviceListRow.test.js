'use strict';
import DeviceListRow from "../DeviceListRow";
import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';

test('DeviceListRow_withDeviceConnected - renders correctly', () => {
    const device = '{"name": "DEVICE_NAME", "id": "DEVICE_ID"}';
    const connectedDeviceIDs = [
        JSON.parse(device).id,
        "DEVICE_ID2",
        "DEVICE_ID3"
    ];

    const tree = renderer.create(
        <DeviceListRow device={device}
                       connectedDeviceIDs={connectedDeviceIDs}
                       devicesTogglingConnection={[]}/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

test('DeviceListRow_withDeviceDisconnected - renders correctly', () => {
    const device = '{"name": "DEVICE_NAME", "id": "DEVICE_ID"}';
    const connectedDeviceIDs = [
        "DEVICE_ID2",
        "DEVICE_ID3"
    ];

    const tree = renderer.create(
        <DeviceListRow device={device}
                       connectedDeviceIDs={connectedDeviceIDs}
                       devicesTogglingConnection={[]}/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});