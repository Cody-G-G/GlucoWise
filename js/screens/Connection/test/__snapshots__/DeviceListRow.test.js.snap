// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceListRow_withDeviceConnected - renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "black",
      "borderWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "flex-start",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "cornflowerblue",
        "color": "white",
        "flex": 4,
        "fontSize": 20,
        "padding": 7,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Name:
    </Text>
     
    DEVICE_NAME
    
    
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Id:
    </Text>
     
    DEVICE_ID
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "firebrick",
        "borderLeftWidth": 1,
        "flex": 2,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 7,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 22,
          "fontWeight": "bold",
        }
      }
    >
      Disconnect
    </Text>
  </View>
</View>
`;

exports[`DeviceListRow_withDeviceDisconnected - renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "black",
      "borderWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "flex-start",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "cornflowerblue",
        "color": "white",
        "flex": 4,
        "fontSize": 20,
        "padding": 7,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Name:
    </Text>
     
    DEVICE_NAME
    
    
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Id:
    </Text>
     
    DEVICE_ID
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "green",
        "borderLeftWidth": 1,
        "flex": 2,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 7,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 22,
          "fontWeight": "bold",
        }
      }
    >
      Connect
    </Text>
  </View>
</View>
`;
